proto/message.py

```diff
@@ -446,6 +478,14 @@ class Message(metaclass=MessageMeta):
     """
 
     def __init__(self, mapping=None, *, ignore_unknown_fields=False, **kwargs):
+
+        self._cached_pb = None
+        # Tracks any new values have had their serialization deferred, and thus whether we
+        # are ready to serialize immediately, or need to sync from the instance back to the
+        # underlying `_pb`
+        self._stale_fields: List[str] = []
```

Comment:
I think `_stale_fields` should be a set instead of a list. It's setting a de-facto sticky dirty bit on a field and then providing iteration.

Reply:
That's a good point. Will change.